cosmwasm/its.ts

```diff
@@ -0,0 +1,172 @@
+import { CosmWasmClient } from '@cosmjs/cosmwasm-stargate';
+import { Command, Option } from 'commander';
+
+import { printError, printInfo } from '../common';
+import { ConfigManager } from '../common/config';
+import { ClientManager, mainProcessor, mainQueryProcessor } from './processor';
+
+export type TokenDataToRegister = {
+    tokenId: string;
+    originChain: string;
+    decimals: number;
+    supply?: string;
+    axelarId: string;
+};
+
+export async function registerToken(
+    interchainTokenServiceAddress: string,
+    client: ClientManager,
+    tokenDataToRegister: TokenDataToRegister,
+    dryRun: boolean,
+) {
+    const supply = tokenDataToRegister.supply;
+    const supplyParam = supply ? { tracked: String(supply) } : 'untracked';
+    const msg = {
+        register_p2p_token_instance: {
+            chain: tokenDataToRegister.axelarId,
+            token_id: formatTokenAddress(tokenDataToRegister.tokenId),
```

Comment:
```suggestion
            token_id: formatTokenId(tokenDataToRegister.tokenId),
```
Token address is a different concept